### PR TITLE
add setpolicy function to kata-runtime tool

### PR DESCRIPTION
**Pull request by LindaYu17**

### CODE REVIEW

**Comment by LindaYu17** [comment]
> @LindaYu17, there's no description, neither in the PR nor in the commits, about what this is trying to achieve and why. Mind to add an explanation, so it becomes easier for the reviewers to understand the reasoning of your changes?

logging/debugging information might probably be disabled in production due to security consideration, but we'd better provide an approach for customer to get logging information during runtime, this PR implement setpolicy function in kata-runtime tools, although it can set whole policy other than logging.
setpolicy would evokes remote attestation, which means before setting policy during runtime, user has to reconfigure new policy hash in KBS/AS.

**Comment by danmihai1** [comment]
@LindaYu17 , would it be reasonable to add to https://github.com/kata-containers/kata-containers/blob/main/docs/design/architecture/README.md#utility-program some example(s) of using this new functionality?

"kata-runtime exec" is very useful for me, so having the ability to do "kata-runtime set_policy" or similar seems useful at a quick look. However, without updating the doc kata-runtime it's not easy for me to use this new functionality.

**Comment by LindaYu17** [comment]
> @LindaYu17 , would it be reasonable to add to https://github.com/kata-containers/kata-containers/blob/main/docs/design/architecture/README.md#utility-program some example(s) of using this new functionality?
> 
> "kata-runtime exec" is very useful for me, so having the ability to do "kata-runtime set_policy" or similar seems useful at a quick look. However, without updating the doc kata-runtime it's not easy for me to use this new functionality.

Thank you Dan. I will add it soon.

**Comment by LindaYu17** [comment]
> > @LindaYu17 , would it be reasonable to add to https://github.com/kata-containers/kata-containers/blob/main/docs/design/architecture/README.md#utility-program some example(s) of using this new functionality?
> > "kata-runtime exec" is very useful for me, so having the ability to do "kata-runtime set_policy" or similar seems useful at a quick look. However, without updating the doc kata-runtime it's not easy for me to use this new functionality.
> 
> Docs are added, thanks.

**Review by danmihai1 — APPROVED** [review]

File: src/runtime/virtcontainers/interfaces.go
Comment (fidencio) [inline src/runtime/virtcontainers/interfaces.go:85]: Identation issue.

**Review by fidencio — COMMENTED** [review]
@LindaYu17, only one of your commits has a commit message.
Mind to also add some explanation to the other 2?

In the future, when debugging something, we'd prefer looking at the commit message to figure out, for instance, why we have to repeat the copy buffer if it's blocked by policy.

Also, in the example, would you mind providing a full example that, later on, could also be used in a test to ensure we will not regress on this?

**Review by fidencio — CHANGES_REQUESTED** [review]
Addiing "Request changes"

File: src/runtime/virtcontainers/interfaces.go
Comment (LindaYu17) [reply src/runtime/virtcontainers/interfaces.go]: fixed, thanks

**Comment by LindaYu17** [comment]
> @LindaYu17, only one of your commits has a commit message. Mind to also add some explanation to the other 2?
> 
> In the future, when debugging something, we'd prefer looking at the commit message to figure out, for instance, why we have to repeat the copy buffer if it's blocked by policy.
> 
> Also, in the example, would you mind providing a full example that, later on, could also be used in a test to ensure we will not regress on this?

more commit message is added.
and a test is also added in last commit.

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM ! Thanks @LindaYu17

File: tests/integration/kubernetes/k8s-runtime-set-policy.bats
Comment (fidencio) [inline tests/integration/kubernetes/k8s-runtime-set-policy.bats:19]: @LindaYu17, we have a problem with this test, as we cannot assume we have `kata-runtime` in the same place where we're running the tests.

For instance, we have tests running on AKS, all the deployment is done in the AKS nodes, and kata-policy is never available for us.

I'd recommend we move this test into a unit test (or similar).

Apart from this, the PR looks good.

File: src/runtime/pkg/containerd-shim-v2/stream.go
Comment (ChengyuZhu6) [inline src/runtime/pkg/containerd-shim-v2/stream.go:133]: @LindaYu17  I recommend verifying whether `err` is null before printing the error log. Otherwise there is a panic: https://github.com/kata-containers/kata-containers/actions/runs/8076580876/job/22066440187?pr=8798#:~:text=panic%3A%20runtime%20error,44

**Review by ChengyuZhu6 — CHANGES_REQUESTED** [review]

File: src/runtime/pkg/containerd-shim-v2/stream.go
Comment (LindaYu17) [reply src/runtime/pkg/containerd-shim-v2/stream.go]: fixed, thanks

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM! thanks @LindaYu17

**Comment by GabyCT** [comment]
@LindaYu17 could you please rebase a fix to the GARM failures has been merged thanks

**Comment by LindaYu17** [comment]
> @LindaYu17 could you please rebase a fix to the GARM failures has been merged thanks

Thank you @GabyCT . I have rebased my PR but there are still two required CI test failures.

**Comment by danmihai1** [comment]
@LindaYu17 @GabyCT - looks like the format of the busybox image has just changed, and exposed a genpolicy bug with that change (https://github.com/kata-containers/kata-containers/actions/runs/8168001754/job/22353019227?pr=8798). I will work on a genpolicy fix asap.

**Comment by danmihai1** [comment]
> @LindaYu17 @GabyCT - looks like the format of the busybox image has just changed, and exposed a genpolicy bug with that change (https://github.com/kata-containers/kata-containers/actions/runs/8168001754/job/22353019227?pr=8798). I will work on a genpolicy fix asap.

Hopefully https://github.com/kata-containers/kata-containers/pull/9243 will unblock CI until we will come up with a better solution.

**Comment by danmihai1** [comment]
@LindaYu17 , when you get a chance please rebase to the latest main commit - to pick up the fix for the cbl-mariner error.

**Comment by ChengyuZhu6** [comment]
All the required tests are passing.

**Pull request merged**
